htlcswitch/interceptable_switch.go

```diff
@@ -394,9 +406,26 @@ func (s *InterceptableSwitch) resolve(res *FwdResolution) error {
 		// Fail with known failure message that is to be encoded and
 		// encrypted.
 		case res.FailureMessage != nil:
+			msg := res.FailureMessage
+
+			// Re-sign the channel update if present. Note that this
+			// changes the passed in FwdResolution.
+			update := getChannelUpdateRef(msg)
+			if update != nil {
+				err := s.validateChannelUpdate(update)
+				if err != nil {
+					return err
+				}
+
+				err = s.resignChannelUpdate(update)
```

Comment:
Should we also commit this update to disk as well? Apologies if this was already an older review comment...

Reply:
No, because the real node isn't aware of this virtual channel at all. It's all up to the external interceptor logic to keep track of the virtual channel properties. As mentioned in https://github.com/lightningnetwork/lnd/pull/7067#discussion_r1286817930, the external interceptor is taking over part of the responsibilities of the switch.